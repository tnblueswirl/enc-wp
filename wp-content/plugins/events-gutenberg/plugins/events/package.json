{
  "name": "@moderntribe/events",
  "version": "0.2.4-alpha",
  "description": "A new WordPress editor experience",
  "repository": "git+https://github.com/moderntribe/events-gutenberg.git",
  "author": "Modern Tribe",
  "license": "GPL-2.0-or-later",
  "private": true,
  "keywords": [
    "WordPress",
    "editor",
    "The Events Calendar"
  ],
  "engines": {
    "node": ">=8.0.0",
    "npm": ">=5.0.0"
  },
  "scripts": {
    "gettext-strings": "cross-env BABEL_ENV=gettext NODE_ENV=production webpack",
    "build": "cross-env NODE_ENV=production webpack -p --progress",
    "dev": "cross-env NODE_ENV=development webpack -d --watch",
    "bootstrap": "./scripts/linkDependencies",
    "preanalyze": "cross-env NODE_ENV=production webpack -p --profile --json > stats.json",
    "analyze": "webpack-bundle-analyzer -m static stats.json"
  },
  "dependencies": {
  }
}
