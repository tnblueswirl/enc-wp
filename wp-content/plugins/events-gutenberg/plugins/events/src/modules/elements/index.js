export { default as Input } from './input/element';
export { default as TermsList } from './terms-list/element';
export { default as OrganizerForm } from './organizer-form/element';
export { default as VenueForm, toFields, toVenue } from './venue-form/element';
export { default as DatePicker } from './date-picker/element';
export { default as TimePicker } from './time-picker/element';
export { default as MetaGroup } from './meta-group/element';
export { default as SearchPosts } from './search-posts/element';
export { default as GoogleMap } from './google-map/element';
export { default as Dashboard } from './dashboard/element';
export { default as CheckBox } from './checkbox/element';
export { default as Month } from './month/element';
export { default as SearchOrCreate } from './search-or-create/element';
export { default as Loading } from './loading';
export { default as YearMonthForm } from './year-month-form/element';
export { default as DateInput } from './date-input/element';
export { default as Upsell } from './upsell/element';
export { default as TimeZone } from './timezone/element';
export { default as EditLink } from './edit-link/element';

import * as Icons from './icons/element';
export { Icons };
