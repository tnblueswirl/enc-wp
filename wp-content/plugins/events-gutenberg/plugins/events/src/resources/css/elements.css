input.tribe-editor--valid{border:1px solid #e62f2f}input.tribe-editor--valid:focus{border:1px solid #e62f2f;outline:none}
.tribe-editor__terms--empty{color:#8d949b}.tribe-editor__terms__label{display:block;font-size:16px}.tribe-editor__terms__list{list-style:none}.tribe-editor__terms__list-item{display:inline-block;margin-right:4px}.tribe-editor__terms__list-item-link{color:#009fd4}.tribe-editor__terms .spinner{display:block;float:none}
.tribe-editor__venue__form{padding:23px 20px;background-color:#f8f9f9;border:1px solid #e0e5e9;border-top:none}.tribe-editor__venue__form h3{font-size:21px;font-size:1.3125rem;margin-bottom:14px;font-weight:700;font-family:Helvetica,Arial,"sans-serif"}.tribe-editor__venue__form dl dt{-ms-flex:0 0 35%;flex:0 0 35%}.tribe-editor__venue__form dl dd{-ms-flex:0 0 65%;flex:0 0 65%}.tribe-editor__venue__fields{width:58%}.tribe-editor__venue__fields input[type=tel].medium,.tribe-editor__venue__fields input[type=text].medium,.tribe-editor__venue__fields input[type=url].medium,.tribe-editor__venue__fields select.medium{width:56.25%}.tribe-editor__venue__fields .row{display:-ms-flexbox;display:flex;-ms-flex-pack:justify;justify-content:space-between}.tribe-editor__venue__fields input[type=tel],.tribe-editor__venue__fields input[type=text],.tribe-editor__venue__fields input[type=url],.tribe-editor__venue__fields select{width:100%;background-color:#fff;border:1px solid #e0e5e9;border-radius:4px;margin-bottom:10px;padding:10px;font-size:16px;font-size:1rem;color:#86909b;font-family:Helvetica,Arial,"sans-serif"}.tribe-editor__venue__fields input[type=tel]:focus,.tribe-editor__venue__fields input[type=text]:focus,.tribe-editor__venue__fields input[type=url]:focus,.tribe-editor__venue__fields select:focus{color:#545d66}.tribe-editor__venue__fields input[type=tel].small,.tribe-editor__venue__fields input[type=text].small,.tribe-editor__venue__fields input[type=url].small,.tribe-editor__venue__fields select.small{width:40.625%}select.tribe-editor__venue__select{height:40px;margin-bottom:10px}
.tribe-editor__datepicker{display:inline-block}
.tribe-editor__timepicker{display:inline-block}.tribe-editor__timepicker-label-container{background-color:#fff;border:2px solid #eaebec;padding-left:10px;border-radius:2px;display:-ms-flexbox;display:flex;-ms-flex-align:center;align-items:center;-ms-flex-pack:center;justify-content:center}.tribe-editor__timepicker-label-container input[type=time]{padding:6px 0;border:none}.tribe-editor__timepicker-label-container input[type=time]::-webkit-clear-button{-webkit-appearance:none;display:none}.tribe-editor__timepicker-label-container button{background-color:#fefffe;color:#545d66;font-family:Helvetica,sans-serif;font-weight:400;line-height:1.5;padding:0 10px;text-decoration:none;border:none;display:-ms-flexbox;display:flex;-ms-flex-align:center;align-items:center;-ms-flex-pack:center;justify-content:center}.tribe-editor__timepicker-label-container .tribe-editor__timepicker__all-day-btn{padding:6px 0}.tribe-editor__timepicker__dialog .tribe-editor__timepicker__item:not(:disabled):hover{background-color:#e7f5fa}.tribe-editor__timepicker__dialog .tribe-editor__timepicker__item:not(:disabled).is-active,.tribe-editor__timepicker__dialog .tribe-editor__timepicker__item:not(:disabled):active,.tribe-editor__timepicker__dialog .tribe-editor__timepicker__item:not(:disabled):focus{position:relative}.tribe-editor__timepicker__dialog .tribe-editor__timepicker__items{height:250px;overflow:auto}.tribe-editor__timepicker__dialog .tribe-editor__timepicker__items .components-placeholder{height:inherit}.tribe-editor__timepicker__dialog .tribe-editor__timepicker__item{display:block;width:100%;color:#555d66;padding:5px 12px;cursor:pointer;border:0;border-bottom:1px solid #e2e4e7;line-height:20px;background:transparent;text-align:left}.tribe-editor__timepicker__dialog .tribe-editor__timepicker__item--current,.tribe-editor__timepicker__dialog .tribe-editor__timepicker__item--current:not(:disabled):hover{color:#fff;background-color:#009fd4}.tribe-editor__timepicker__dialog.components-popover .components-popover__content{min-width:110px}
.tribe-editor__search-posts{width:340px}.tribe-editor__search-posts .tribe-editor__search-posts__results-list-item-button.is-active,.tribe-editor__search-posts .tribe-editor__search-posts__results-list-item-button:active,.tribe-editor__search-posts .tribe-editor__search-posts__results-list-item-button:focus{position:relative}.tribe-editor__search-posts .editor-inserter__search{width:100%}.tribe-editor__search-posts .tribe-editor__search-posts__results{height:250px;overflow:auto}.tribe-editor__search-posts .tribe-editor__search-posts__results .components-placeholder{height:inherit}.tribe-editor__search-posts .tribe-editor__search-posts__results-list,.tribe-editor__search-posts .tribe-editor__search-posts__results-list-item{margin:0}.tribe-editor__search-posts .tribe-editor__search-posts__results-list-item-button{display:block;width:100%;color:#555d66;padding:12px;cursor:pointer;border:0;border-bottom:1px solid #e2e4e7;line-height:20px;background:transparent;text-align:left}.tribe-editor__search-posts .tribe-editor__search-posts__results-list-item-button:hover{background-color:#f8f9f9}.tribe-editor__dropdown{display:inline-block}.tribe-editor__spinner-container.tribe-editor__spinner--item{padding:0}
.tribe-editor__map--dynamic{width:100%;height:100%;z-index:2}.tribe-editor__map--interactive{width:100%;display:-ms-flexbox;display:flex;-ms-flex-align:center;align-items:center;-ms-flex-pack:center;justify-content:center;height:100%;-ms-flex:1;flex:1}.tribe-editor__map{background-color:#f7f1e1}.tribe-editor__venue--has-map .tribe-editor__map{width:49%;-ms-flex-preferred-size:49%;flex-basis:49%;min-height:270px}.tribe-editor__map--static{display:-ms-flexbox;display:flex;-ms-flex-align:center;align-items:center;-ms-flex-pack:center;justify-content:center}.tribe-editor__map--static img{z-index:2}
.tribe-editor__dashboard:after{height:20px;left:0;right:0;position:absolute;content:"";bottom:-20px}.tribe-editor__dashboard__container{pointer-events:none;display:none;position:relative}.tribe-editor__dashboard__container--open{display:block;pointer-events:auto;width:100%}.tribe-editor__dashboard{background-color:#f8f9f9;border:1px solid #e3e4e8;left:0;margin-left:-29px;margin-right:-29px;position:absolute;right:0;z-index:100}.tribe-editor__dashboard .tribe-editor__time-picker__separator{color:#545d66;margin-left:8px;margin-right:6px}.tribe-editor__dashboard .tribe-editor__time-picker__label{color:#545d66;font-family:Helvetica,Arial,sans-serif;font-size:1em;font-weight:400;margin-right:8px}.tribe-editor__dashboard__container--up .tribe-editor__dashboard{bottom:-14px}.tribe-editor__dashboard__container--down .tribe-editor__dashboard{top:14px}.tribe-editor__dashboard:before{content:"";display:block;height:50px;top:-34px;left:0;right:0;position:absolute;z-index:2}
.tribe-editor__checkbox{display:-ms-flexbox;display:flex;-ms-flex-align:center;align-items:center;-ms-flex-pack:center;justify-content:center;background-color:transparent;border:none}.tribe-editor__checkbox label{font-family:Helvetica,"sans-serif";color:#545d66;margin-right:14px}.tribe-editor__checkbox input[type=checkbox]{display:none}.tribe-editor__checkbox svg{pointer-events:none}
.DayPicker{display:inline-block}.DayPicker-wrapper{position:relative;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;padding-bottom:16px;padding-bottom:1rem;-ms-flex-direction:row;flex-direction:row}.DayPicker-Months{display:-ms-flexbox;display:flex;-ms-flex-wrap:wrap;flex-wrap:wrap;-ms-flex-pack:center;justify-content:center}.DayPicker-Month{display:table;border-collapse:collapse;border-spacing:0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;margin:0 16px;margin:0 1rem;margin-top:16px;margin-top:1rem}.DayPicker-NavButton{position:absolute;cursor:pointer;top:16px;top:1rem;right:24px;right:1.5rem;margin-top:2px;color:#8b9898;width:20px;width:1.25rem;height:20px;height:1.25rem;display:inline-block;background-size:50%;background-repeat:no-repeat;background-position:50%}.DayPicker-NavButton:hover{opacity:.8}.DayPicker-NavButton--prev{margin-right:24px;margin-right:1.5rem;background-image:url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAwCAYAAAB5R9gVAAAABGdBTUEAALGPC/xhBQAAAVVJREFUWAnN2G0KgjAYwPHpGfRkaZeqvgQaK+hY3SUHrk1YzNLay/OiEFp92I+/Mp2F2Mh2lLISWnflFjzH263RQjzMZ19wgs73ez0o1WmtW+dgA01VxrE3p6l2GLsnBy1VYQOtVSEH/atCCgqpQgKKqYIOiq2CBkqtggLKqQIKgqgCBjpJ2Y5CdJ+zrT9A7HHSTA1dxUdHgzCqJIEwq0SDsKsEg6iqBIEoq/wEcVRZBXFV+QJxV5mBtlDFB5VjYTaGZ2sf4R9PM7U9ZU+lLuaetPP/5Die3ToO1+u+MKtHs06qODB2zBnI/jBd4MPQm1VkY79Tb18gB+C62FdBFsZR6yeIo1YQiLJWMIiqVjQIu1YSCLNWFgijVjYIuhYYCKoWKAiiFgoopxYaKLUWOii2FgkophYp6F3r42W5A9s9OcgNvva8xQaysKXlFytoqdYmQH6tF3toSUo0INq9AAAAAElFTkSuQmCC")}.DayPicker-NavButton--next{background-image:url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAwCAYAAAB5R9gVAAAABGdBTUEAALGPC/xhBQAAAXRJREFUWAnN119ugjAcwPHWzJ1gnmxzB/BBE0n24m4xfNkTaOL7wOtsl3AXMMb+Vjaa1BG00N8fSEibPpAP3xAKKs2yjzTPH9RAjhEo9WzPr/Vm8zgE0+gXATAxxuxtqeJ9t5tIwv5AtQAApsfT6TPdbp+kUBcgVwvO51KqVhMkXKsVJFXrOkigVhCIs1Y4iKlWZxB1rX4gwlpRIIpa8SDkWmggrFq4IIRaJKCYWnSgnrXIQV1r8YD+1Vrn+bReagysIFfLABRt31v8oBu1xEBttfRbltmfjgEcWh9snUS2kNdBK6WN1vrOWxObWsz+fjxevsxmB1GQDfINWiev83nhaoiB/CoOU438oPrhXS0WpQ9xc1ZQWxWHqUYe0I0qrKCQKjygDlXIQV2r0IF6ViEBxVTBBSFUQQNhVYkHIVeJAtkNsbQ7c1LtzP6FsObhb2rCKv7NBIGoq4SDmKoEgTirXAcJVGkFSVVpgoSrXICGUMUH/QBZNSUy5XWUhwAAAABJRU5ErkJggg==")}.DayPicker-NavButton--interactionDisabled{display:none}.DayPicker-Caption{padding:0 8px;padding:0 .5rem;display:table-caption;text-align:left;margin-bottom:8px;margin-bottom:.5rem}.DayPicker-Caption>div{font-size:18.4px;font-size:1.15rem;font-weight:500}.DayPicker-Weekdays{margin-top:16px;margin-top:1rem;display:table-header-group}.DayPicker-WeekdaysRow{display:table-row}.DayPicker-Weekday{display:table-cell;padding:8px;padding:.5rem;font-size:.875em;text-align:center;color:#8b9898}.DayPicker-Weekday abbr[title]{border-bottom:none;text-decoration:none}.DayPicker-Body{display:table-row-group}.DayPicker-Week{display:table-row}.DayPicker-Day{text-align:center;border-radius:50%}.DayPicker-Day,.DayPicker-WeekNumber{display:table-cell;padding:8px;padding:.5rem;cursor:pointer;vertical-align:middle}.DayPicker-WeekNumber{text-align:right;min-width:16px;min-width:1rem;font-size:.75em;color:#8b9898;border-right:1px solid #eaecec}.DayPicker--interactionDisabled .DayPicker-Day{cursor:default}.DayPicker-Footer{padding-top:8px;padding-top:.5rem}.DayPicker-TodayButton{border:none;background-image:none;background-color:transparent;box-shadow:none;cursor:pointer;color:#4a90e2;font-size:.875em}.DayPicker-Day--today{color:#d0021b;font-weight:700}.DayPicker-Day--outside{cursor:default;color:#8b9898}.DayPicker-Day--disabled{color:#dce0e0;cursor:default}.DayPicker-Day--sunday{background-color:#f7f8f8}.DayPicker-Day--sunday:not(.DayPicker-Day--today){color:#dce0e0}.DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside){position:relative;color:#f0f8ff;background-color:#4a90e2}.DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside):hover{background-color:#51a0fa}.DayPicker:not(.DayPicker--interactionDisabled) .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover{background-color:#f0f8ff}.DayPickerInput{display:inline-block}.DayPickerInput-OverlayWrapper{position:relative}.DayPickerInput-Overlay{left:0;z-index:1;position:absolute;background:#fff;box-shadow:0 2px 5px rgba(0,0,0,.15)}
.tribe-editor__calendars .DayPicker{width:100%}.tribe-editor__calendars .DayPicker-Months{-ms-flex-pack:justify;justify-content:space-between}.tribe-editor__calendars .DayPicker-Month{width:46.2686567164%;margin:0}.tribe-editor__calendars .DayPicker-Caption>div,.tribe-editor__calendars .DayPicker-Day,.tribe-editor__calendars .DayPicker-Weekday{color:#545d66;font-family:Helvetica,"sans-serif";font-weight:400}.tribe-editor__calendars .DayPicker-Caption>div{font-size:16px;font-size:1rem;margin-bottom:12px;text-align:center}.tribe-editor__calendars .DayPicker-Weekday{font-size:12px;font-size:.75rem}.tribe-editor__calendars .DayPicker-Day{font-size:14px;font-size:.875rem}.tribe-editor__calendars .DayPicker-Day:hover{color:#007bb4;background-color:#fff}.tribe-editor__calendars .DayPicker-Day--today{color:#545d66}.tribe-editor__calendars .DayPicker-Day--disabled{pointer-events:none;color:#ccc}.tribe-editor__calendars .DayPicker-Day--selected:not(.DayPicker-Day--outside){border-radius:0;background-color:#009fd4;color:#fff}.tribe-editor__calendars .DayPicker-Day--selected:not(.DayPicker-Day--outside):hover{background-color:#007bb4;color:#fff}.tribe-editor__calendars .DayPicker-NavButton--prev{left:0;top:0}.tribe-editor__calendars .DayPicker-NavButton--next{right:0;top:0}.tribe-editor__calendars .DayPicker:not(.DayPicker--interactionDisabled) .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover{background-color:#fff}.tribe-editor__daypicker-caption{padding:0 8px;display:table-caption;margin-bottom:8px}.tribe-editor__daypicker-caption>div{font-size:16px;font-size:1rem;margin-bottom:12px;text-align:center;color:#545d66;font-family:Helvetica,"sans-serif";font-weight:400}.tribe-editor__calendars--range .DayPicker-Day--selected:not(.DayPicker-Day--start):not(.DayPicker-Day--end):not(.DayPicker-Day--outside){background-color:#e7f5fa;color:#545d66}.tribe-editor__calendars--range .DayPicker-Day--selected:not(.DayPicker-Day--start):not(.DayPicker-Day--end):not(.DayPicker-Day--outside):hover{color:#007bb4}
.tribe-editor__soc__input:focus{outline:none}.tribe-editor__soc__input{border:none;padding:16px 11px 16px 0;font-size:18px;font-size:1.125rem;font-family:Helvetica,Arial,"sans-serif";font-weight:700;color:#86909b;-ms-flex:1;flex:1;margin:0}.tribe-editor__soc__input:-ms-input-placeholder{font-weight:400;color:#86909b}.tribe-editor__soc__input::placeholder{font-weight:400;color:#86909b}.tribe-editor__soc__input__container{border:1px solid #e0e5e9;border-radius:4px;width:100%;display:-ms-flexbox;display:flex;-ms-flex-align:center;align-items:center;-ms-flex-wrap:wrap;flex-wrap:wrap}.tribe-editor__soc__input__container svg{margin-left:14px;margin-right:14px;fill:#e2e4e7;width:22px}.tribe-editor__soc__input__container path{fill:#e2e4e7}.tribe-editor__soc__input__container--active{border:1px solid #e0e5e9;border-bottom-left-radius:0;border-bottom-right-radius:0;border-right:none}.tribe-editor__soc__input__container--active path,.tribe-editor__soc__input__container--active svg{fill:#009fd4}.tribe-editor__soc__input__container--active .tribe-editor__soc__input{border-right:1px solid #e0e5e9;color:#000}.tribe-editor__soc__results--loading,ul.tribe-editor__soc__results{border:1px solid #cdd3d8;background-color:#fafafa;box-shadow:1px 1px 6px rgba(0,0,0,.1);border-bottom-left-radius:8px;border-bottom-right-radius:8px;padding:18px 28px;width:100%}.tribe-editor__soc__results--loading{display:-ms-flexbox;display:flex;-ms-flex-align:center;align-items:center;-ms-flex-pack:center;justify-content:center}.tribe-editor__soc__results li:last-child{margin-bottom:0}.tribe-editor__soc__results strong{color:#000}.tribe-editor__soc__results li{list-style:none;padding:10px 15px;margin-bottom:0;color:#67686c;cursor:pointer;font-family:Helvetica,Arial,"sans-serif"}.tribe-editor__soc__results li:hover{color:#009fd4}
.tribe-editor__spinner-container{display:-ms-flexbox;display:flex;width:100%;-ms-flex-pack:center;justify-content:center;-ms-flex-align:center;align-items:center;padding:20px}
.tribe-editor__year-month-form{display:table-caption;text-align:center;margin-bottom:18px}.tribe-editor__year-month-form .tribe-editor__year-month-form__month,.tribe-editor__year-month-form .tribe-editor__year-month-form__year{font-size:16px;border:none;box-shadow:0 1px 0 rgba(0,0,0,.1)}
.tribe-editor__date-input{width:100%}.tribe-editor__date-input input[type=text]{width:100%;padding:10px}
.tribe-editor__subtitle__footer-upsell{padding:15px 22px;background-color:#e7f5fa;border-top:1px solid #e3e4e8}p.tribe-editor__subtitle__footer-upsell-text{font-size:16px;font-style:normal;font-family:Helvetica,sans-serif;font-weight:400;line-height:1.3;color:#191e22;margin:0}.tribe-editor__subtitle__footer-upsell-link{color:#191e22}
.tribe-editor__timezone-input input{border:0;padding:2px 5px;font-size:16px}
.tribe-editor__edit-link{color:#009fd4}.tribe-editor__edit-link:hover{color:#027bb4;text-decoration:none}
.editor-block-inspector__card .tribe-editor__icons__container{padding:4px;display:-ms-flexbox;display:flex;-ms-flex-align:center;align-items:center;-ms-flex-pack:center;justify-content:center}.editor-block-inspector__card .tribe-editor__icons__container svg{width:28px;height:28px}.editor-block-inspector__card .tribe-editor__icons--tec{background-color:#199fd1}

/*# sourceMappingURL=elements.css.map*/