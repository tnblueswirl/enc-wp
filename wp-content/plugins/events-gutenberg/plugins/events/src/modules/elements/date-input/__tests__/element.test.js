/**
 * External dependencies
 */
import renderer from 'react-test-renderer';
import React from 'react';

// Prevent to load a mocked version of lodash
jest.unmock( 'lodash' );
const lodash = require.requireActual( 'lodash' );

/**
 * Internal dependencies
 */
import { DateInput } from '@moderntribe/events/elements';

describe( 'DateInput element', () => {
	it( 'Should render the input', () => {
		const component = renderer.create( <DateInput /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the before child on the input', () => {
		const component = renderer.create(
			<DateInput before={ <span>Before the input</span> } />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the after child on the input', () => {
		const component = renderer.create(
			<DateInput after={ <span>After the input</span> } />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the before and after child', () => {
		const component = renderer.create(
			<DateInput
				before={ <span>Before the input</span> }
				after={ <span>After the input</span> }
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should trigger the change on the input field', () => {
		const onChange = jest.fn();
		const component = (
			<DateInput
				onChange={ onChange }
				value="Modern Tribe"
			/>
		);
		const mounted = mount( component );
		const render = renderer.create( component );
		mounted.find( 'input' ).simulate( 'change' );

		expect( render.toJSON() ).toMatchSnapshot();
		expect( onChange ).toHaveBeenCalled();
		expect( onChange ).toHaveBeenCalledWith( 'Modern Tribe' );
	} );

	it( 'Should trigger the setDateTime callback', () => {
		const setDateTime = jest.fn();
		lodash.debounce = ( callback ) => jest.fn( ( ...args ) => callback( ...args ) );
		const component = (
			<DateInput
				setDateTime={ setDateTime }
				value="17 August 2018 - 19 August 2018"
			/>
		);
		const mounted = mount( component );
		const render = renderer.create( component );
		mounted.find( 'input' ).simulate( 'change' );

		const expected = {
			end: '2018-08-19 12:00:00',
			start: '2018-08-17 12:00:00',
		};
		expect( render.toJSON() ).toMatchSnapshot();
		expect( setDateTime ).toHaveBeenCalled();
		expect( setDateTime ).toHaveBeenCalledWith( expected );
	} );
} );
