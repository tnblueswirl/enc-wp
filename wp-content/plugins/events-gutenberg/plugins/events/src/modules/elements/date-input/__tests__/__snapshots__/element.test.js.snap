// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput element Should render the after child on the input 1`] = `
<div
  className="tribe-editor__date-input__container"
>
  <input
    className="tribe-editor__date-input"
    name="date-input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <span>
    After the input
  </span>
</div>
`;

exports[`DateInput element Should render the before and after child 1`] = `
<div
  className="tribe-editor__date-input__container"
>
  <span>
    Before the input
  </span>
  <input
    className="tribe-editor__date-input"
    name="date-input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <span>
    After the input
  </span>
</div>
`;

exports[`DateInput element Should render the before child on the input 1`] = `
<div
  className="tribe-editor__date-input__container"
>
  <span>
    Before the input
  </span>
  <input
    className="tribe-editor__date-input"
    name="date-input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`DateInput element Should render the input 1`] = `
<div
  className="tribe-editor__date-input__container"
>
  <input
    className="tribe-editor__date-input"
    name="date-input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`DateInput element Should trigger the change on the input field 1`] = `
<div
  className="tribe-editor__date-input__container"
>
  <input
    className="tribe-editor__date-input"
    name="date-input"
    onChange={[Function]}
    type="text"
    value="Modern Tribe"
  />
</div>
`;

exports[`DateInput element Should trigger the setDateTime callback 1`] = `
<div
  className="tribe-editor__date-input__container"
>
  <input
    className="tribe-editor__date-input"
    name="date-input"
    onChange={[Function]}
    type="text"
    value="17 August 2018 - 19 August 2018"
  />
</div>
`;
